### PR TITLE
Initial changes for supporting USB MJPEG cameras

### PR DESCRIPTION
**Pull request by chrisruk**
Currently Qt Preview and Null Preview should function with USB MJPEG cameras

Signed-off-by: Chris Richardson <christopher.richardson@raspberrypi.org>

### CODE REVIEW

File: picamera2/picamera2.py [hunk 1]
```diff
@@ -534,8 +533,10 @@ def create_preview_configuration(self, main={}, lores=None, raw=None, transform=
             self.align_stream(lores, optimal=False)
         raw = self._make_initial_stream_config({"format": self.sensor_format, "size": main["size"]}, raw, self._raw_stream_ignore_list)
         # Let the framerate vary from 12fps to as fast as possible.
-        controls = {"NoiseReductionMode": libcamera.controls.draft.NoiseReductionModeEnum.Minimal,
-                    "FrameDurationLimits": (100, 83333)} | controls
+        ctrls = set([ctrl.name for ctrl in self.camera.controls.keys()])
+        if "NoiseReductionMode" in ctrls and "FrameDurationLimits" in ctrls:
+            controls = {"NoiseReductionMode": libcamera.controls.draft.NoiseReductionModeEnum.Minimal,
```
Comment (davidplowman) [inline picamera2/picamera2.py:538]: Maybe
```
if "NoiseReductionMode" in self.camera_controls and "FrameDurationLimits" in self.camera_controls:
```
woudl be a bit easier to read? Ditto for the other two places below where we have this.

File: picamera2/request.py [hunk 2]
```diff
@@ -235,17 +240,21 @@ def save(self, img, metadata, file_output, format=None):
...
-            zero_ifd = {piexif.ImageIFD.Make: "Raspberry Pi",
+            if "AnalogueGain" in metadata and "DigitalGain" in metadata:
+                zero_ifd = {piexif.ImageIFD.Make: "Raspberry Pi",
                         piexif.ImageIFD.Model: self.picam2.camera.id,
                         piexif.ImageIFD.Software: "Picamera2"}
-            total_gain = metadata["AnalogueGain"] * metadata["DigitalGain"]
-            exif_ifd = {piexif.ExifIFD.ExposureTime: (metadata["ExposureTime"], 1000000),
-                        piexif.ExifIFD.ISOSpeedRatings: int(total_gain * 100)}
-            exif = piexif.dump({"0th": zero_ifd, "Exif": exif_ifd})
+                total_gain = metadata["AnalogueGain"] * metadata["DigitalGain"]
```
Comment (davidplowman) [inline picamera2/request.py:247]: I think the line above (and the next few) are complaining about indentation and I suspect it's caused by lines 245 and 246 being insufficiently indented. Do you just want to double-check that? Thanks!

**Comment by davidplowman** [comment]
Once we've merged this one I'll post another teeny commit which will make the YUVV modes work too (well, up to a point, this whole USB camera thing is a bit "best efforts").

**Comment by chrisruk** [comment]
Cool, sounds good!

**Pull request merged**